src/Controller/RenderPluginInstancePreviewApiController.php

```diff
@@ -25,33 +25,24 @@ public function __construct(PluginManager $pluginManager)
 
     public function indexAction()
     {
-        // This does not work
-        $pluginType = $this->params()->fromPost('pluginType');
-        $instanceId = $this->params()->fromPost('instanceId');
-        $instanceConfig = $this->params()->fromPost('instanceConfig');
-
         $data = json_decode(
             $this->getRequest()->getContent(),
             true
         );
 
-        if (json_last_error() !== JSON_ERROR_NONE) {
+        if (json_last_error() !== JSON_ERROR_NONE
+            || !is_array($data)
+            || !array_key_exists('pluginType', $data)
+            || !array_key_exists('instanceId', $data)
+            || !array_key_exists('instanceConfig', $data)
+        ) {
             $response = new Response();
+            $response->setContent('400 Bad Request');
             $response->setStatusCode(400);
 
             return $response;
```

Comment:
would be useful if you could give back something like `{"error": "<reason for failure>"}` instead of empty response